web/src/app/modules/views/main/editors/modules/contents-container/components/test-specification-container.component.ts

```diff
@@ -0,0 +1,67 @@
+import { Component, Input } from '@angular/core';
+import { ContentContainerBase } from '../base/contents-container-base';
+import { IContainer } from '../../../../../../../model/IContainer';
+import { ModelFactoryBase } from '../../../../../../../factory/model-factory-base';
+import { CEGModelFactory } from '../../../../../../../factory/ceg-model-factory';
+import { Type } from '../../../../../../../util/type';
+import { Process } from '../../../../../../../model/Process';
+import { TestSpecification } from '../../../../../../../model/TestSpecification';
+import { TestSpecificationFactory } from '../../../../../../../factory/test-specification-factory';
+import { ElementFactoryBase } from '../../../../../../../factory/element-factory-base';
+import { Sort } from '../../../../../../../util/sort';
+import { SpecmateDataService } from '../../../../../../data/modules/data-service/services/specmate-data.service';
+import { NavigatorService } from '../../../../../../navigation/modules/navigator/services/navigator.service';
+import { TranslateService } from '@ngx-translate/core';
+import { ConfirmationModal } from '../../../../../../notification/modules/modals/services/confirmation-modal.service';
+import { UUID } from 'angular2-uuid';
+import { Id } from '../../../../../../../util/id';
+import { TestCase } from '../../../../../../../model/TestCase';
+
+@Component({
+    moduleId: module.id.toString(),
+    selector: 'test-specification-container',
+    templateUrl: 'test-specification-container.component.html',
+    styleUrls: ['test-specification-container.component.css']
+})
+
+export class TestSpecificationContainer extends ContentContainerBase<TestSpecification> {
+
+    constructor(dataService: SpecmateDataService, navigator: NavigatorService, translate: TranslateService, modal: ConfirmationModal) {
+        super(dataService, navigator, translate, modal);
+    }
+
+    private _parent: IContainer;
+
+    protected get parent(): IContainer {
+        return this._parent;
+    }
+
+    @Input()
+    protected set parent(parent: IContainer) {
+        this._parent = parent;
+        this.readContents();
+    }
+
+    public newName: string;
+
+    protected condition = (element: IContainer) => Type.is(element, TestSpecification);
+
+    public async createElement(name: string): Promise<TestSpecification> {
+        let factory: ElementFactoryBase<TestSpecification> = new TestSpecificationFactory(this.dataService);
+        return await factory.create(this.parent, true, Id.uuid, name);
```

Comment:
in the ther classes you store the result in a variable and return with a cast. Any reason it is not done like this here?

Reply:
In the other classes, we need to do the cast, because the factory used there returns something of the type ```CEGModel | Process```. To return the correct type here in these methods, which return either a ```CEGModel``` or a ```Process```, we need to do the cast. But the factory used here in this method, already has the correct return type in the create method.